plugins/woocommerce/includes/widgets/class-wc-widget-cart.php

```diff
@@ -43,6 +43,44 @@ public function __construct() {
 		parent::__construct();
 	}
 
+	/**
+	 * Get list of Woo blocks that update the cart.
+	 *
+	 * @return array;
+	 */
+	public static function get_blocks() {
+		return array(
+			'woocommerce/handpicked-products',
+			'woocommerce/single-product',
+			'woocommerce/product-collection',
+			'woocommerce/add-to-cart-form',
+			'woocommerce/product-top-rated',
+			'woocommerce/product-tag',
+			'woocommerce/products-by-attribute',
+			'woocommerce/product-on-sale',
+			'woocommerce/product-new',
+			'woocommerce/product-category',
+			'woocommerce/product-best-sellers',
+			'woocommerce/all-products',
+		);
+	}
+
+	/**
+	 * array_contains
+	 * Check if string contains word from array.
+	 *
+	 * @since 1.0.0
+	 * @version 1.0.0
+	 **/
+	function array_contains( $str, array $arr ) {
```

Comment:
It would be nice to reshape this to resemble `has_block`, so something like `has_blocks( srtring[] $blocks, int|string|WP_Post|null $post)`

Reply:
I added a fix in this [commit](https://github.com/woocommerce/woocommerce/pull/38910/commits/dc47b563c013755ae85e797419449aa627002e84). I named the function `has_block_from_list` since `has_blocks` already exist!